### PR TITLE
dockerbuild update

### PR DESCRIPTION
**Pull request by JudahSchwartz**
circleci hasn't updated their images in over two years. recreated the image based on latest python 3.10 debian
Fixes: https://jira-dc.paloaltonetworks.com/browse/CIAC-8984

commit that tested several images https://github.com/demisto/dockerfiles/pull/22135/commits/fa08be4675ef3277d8a0be8bbf6d8705f52f9d7f

### CODE REVIEW

File: docker/build_docker.sh [hunk 1]
```diff
@@ -293,11 +293,6 @@ function docker_build {
         echo "Skipping license verification for devonly image"
     else
         PY3CMD="python3"
-        if command -v python3.7 >/dev/null 2>&1; then
```
Comment (JudahSchwartz) [inline docker/build_docker.sh:296]: no reason to support running this scripts from old python

File: docker/dockerbuild/Dockerfile [hunk 2]
```diff
@@ -1,23 +1,38 @@
-# Last modified: 2023-01-04T00:13:10.301105+00:00
-FROM circleci/python:3.10.1
-
+# Last modified: 2023-11-23T00:13:10.301105+00:00
+FROM python:3.10.13-slim-bookworm
+COPY requirements.txt .
+COPY localtime /etc/localtime
+RUN set -ex; apt-get update; apt-get install -y --no-install-recommends autoconf automake bzip2 dpkg-dev file g++ gcc  \
```
Comment (JudahSchwartz) [inline docker/dockerbuild/Dockerfile:5]: mostly taken from circleci layers

**Review by glicht — APPROVED** [review]
Looks good

File: docker/dockerbuild/Dockerfile [hunk 3]
```diff
@@ -1,23 +1,33 @@
...
-# Setup pyenv + poetry and install extra python versions
-RUN curl -L https://github.com/pyenv/pyenv-installer/raw/master/bin/pyenv-installer | bash \
-  && curl -sSL https://install.python-poetry.org | python3 \
-  && poetry --version \
-  && poetry self add poetry-plugin-export \
-  && echo 'export PATH="/home/circleci/.pyenv/bin:$PATH"' >> ~/.bashrc \
-  && echo 'eval "$(pyenv init -)"' >> ~/.bashrc \
-  && echo 'eval "$(pyenv virtualenv-init -)"' >> ~/.bashrc
+USER root
+# install docker
```
[the remark this answers is not in the document]
Comment (JudahSchwartz) [reply docker/dockerbuild/Dockerfile:5]: this upgrades the docker version to latest. image comes with version 20.xx, latest is 24.xx

**Pull request merged**